document updated to include playwright test secret information (#198)
DFE-Digital/find-information-about-academies-and-trusts: docs/adrs/0003-authorization-bypass-for-ui-autotests.md

# 3. Authorization bypass for UI auto tests

Date: 2023-10-26

## Status

Accepted

## Context

Up until now, we have been unable to run ui automation tests on dev or test enviroments.

Due to DfE's policy of only allowing certain IP ranges access to authenticate via microsoft, test users will not work in the test pipelines in the same way as a normal user so we needed to implement a special route to enable the test user to bypass authentication in non-prod environments.

## Decision

> We will implement the dotnet class DenyAnonymousAuthorizationRequirement which allow bypassing of authorization functionality.

Reviewer: ```suggestion
We will implement the dotnet class `AuthorizationHandler<DenyAnonymousAuthorizationRequirement>` which allows bypassing of our standard authorization route.
```

> It requires a secret guid to be stored in azure which is passed in an authorization header with every test automation request.

Reviewer: ```suggestion
It requires a secret to be stored in azure which is passed in an authorization header with every test automation request.
```
We don't want to expose the form of the secret to people reading the docs to look for security holes. Also we're no longer storing it as a GUID

The bypass is only allowed on non production environments.

> This method as been used in other Dfe projects within the program.

Reviewer: ```suggestion
This method has been used in other DfE projects within the program.
```

> ## Consequences

Reviewer: More consequences:

- Developers will no longer be able to access a locally running CI docker instance through their browser because the header will be missing from their browser requests
- Integration and Deployment tests can be reenabled in the pipeline. The security test pipeline will also now function on demand
